docs/knowledge-base/jobnik/README.md

```diff
@@ -0,0 +1,137 @@
+
+[Link to Job Manager Repository](https://github.com/MapColonies/jobnik-manager)
+
+---
+
+## 2. Jobnik SDK
+
+The Jobnik SDK is a TypeScript library that abstracts the complexity of communicating with the Job Manager. It provides a typed, resilient communication layer used by both producers (services that create work) and consumers (services that perform work).
+
+**Key Components:**
+
+- **Producer Client:** A client designed for services that initiate workflows. It handles the creation of Jobs, Stages, and Tasks, automatically managing trace context propagation.
+- **Worker Client:** A client designed for task processing services. It manages the task lifecycle loop: polling for tasks, executing business logic, and updating task status (Completed/Failed) upon conclusion.
+
+**Core Capabilities:**
+
+- **Type Safety:** Leverages TypeScript Generics to allow developers to define custom schemas for Job, Stage, and Task data. This ensures that payloads passed between Producers and Workers are typed and validated at compile time.
+- **Resilience:** Built-in implementation of industry-standard stability patterns, including:
+  - **Circuit Breakers:** Prevents cascading failures when the Job Manager is under load.
+  - **Exponential Backoff:** Manages retry logic for polling and network requests.
+  - **Graceful Shutdown:** Handles signal termination to ensure active tasks are completed or released before the service stops.
+- **Automated Observability:** Automatically injects tracing headers into API calls and exposes worker-specific metrics (e.g., active tasks, task duration, success rates) without requiring manual instrumentation in the business logic.
+
+[Link to Jobnik SDK](/docs/knowledge-base/packages/jobnik-sdk/README.md)
+
+---
+
+## 3. Worker Boilerplate
+
+The Worker Boilerplate is a production-ready template repository designed to accelerate the development of new task-processing services. While not a runtime dependency, it serves as the standard starting point for building services that utilize the Jobnik SDK.
+
+**Purpose:**
+
+It provides a pre-configured environment that integrates the Jobnik SDK with standard infrastructure requirements, allowing developers to focus solely on implementing the specific business logic for a task (e.g., image processing, data ingestion).
+
+**Included Scaffolding:**
+
+- **Dependency Injection:** Pre-wired setup using `tsyringe` for managing service dependencies.
+- **Configuration Management:** Integration with standard configuration libraries for environment-based settings.
+- **SDK Integration:** Example implementation of the SDK's `Worker` client, demonstrating how to register task handlers and define custom types.
+- **Deployment Assets:** Includes Dockerfiles and Helm charts aligned with infrastructure standards for rapid deployment.
+
+[Link to Worker Boilerplate Repository](/docs/knowledge-base/jobnik/boilerplate/README.md)
+
```

Comment:
should mention:
"The worker boilerplate represents the best practice for implementation; using the SDK directly with existing workers is an alternative, but not the preferred rational choice.

Reply:
mention like in the sdk that its dedicated for typescript develop